src/components/accounts/two_factor/TwoFactorVerifyModal.js

```diff
@@ -0,0 +1,96 @@
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { useDispatch, useSelector } from 'react-redux';
+import Modal from "../../common/modal/Modal";
+import ModalTheme from '../ledger/ModalTheme';
+import MobileActionSheet from '../../common/modal/MobileActionSheet';
+import FormButton from '../../common/FormButton';
+import { Translate } from 'react-localize-redux';
+import TwoFactorVerifyInput from './TwoFactorVerifyInput';
+import { verifyTwoFactor, clearAlert, resendTwoFactor } from '../../../actions/account';
+import { wallet } from '../../../utils/wallet';
+
+const Form = styled.form`
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    width: 100%;
+`
+
+const TwoFactorVerifyModal = ({ open, onClose }) => {
+
+    const [method, setMethod] = useState();
+    const [code, setCode] = useState('');
+    const dispatch = useDispatch();
+    const account = useSelector(({ account }) => account);
+    const loading = account.actionsPending.includes('VERIFY_TWO_FACTOR');
+    const error = account.requestStatus && account.requestStatus.messageCode === 'account.verifyTwoFactor.error';
+    // console.log(account)
+
+    useEffect(() => {
+        let isMounted = true;
+
+        const handleGetTwoFactor = async () => {
+            setMethod(await wallet.get2faMethod())
+        };
+
+        if (isMounted) {
+            handleGetTwoFactor()
+        }
+        
+        return () => { isMounted = false }
```

Comment:
Not sure why is there a need to keep track of whether component is "mounted"? If the goal is to call this action once per 2FA request I think following makes sense:
1) Pass 2FA request or it's unique identifier as a prop to this component.
2) Pass it as a dependency to `useEffect`

Reply:
I kept seeing the `Warning: Can't perform a React state update on an unmounted component.` if the component unmounted before the async function finished so I used the isMounted flag to keep track of the status. Any idea how to handle that better?

Reply:
Got it, I see now it's common pattern.